### PR TITLE
Improve types

### PR DESCRIPTION
**Pull request by martijngastkemper**

### CODE REVIEW

File: functions/compose.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *
  * Takes n function arguments.
  *
- * @param  callable[] ...$args
+ * @param  callable|callable-string ...$args
```
Comment (harmenjanssen) [inline functions/compose.php:21]: How does this work with method calls? I'm assuming `callable-string` cannot contain `[$myObject, "myMethod"]`.

**Review by harmenjanssen — APPROVED** [review]
One question about `callable-string`, but I will already approve this one! Thanks!

File: functions/compose.php (hunk 1 above)
Comment (martijngastkemper) [reply functions/compose.php:21]: I've added a unit test and PHPStan checks those classes. It doesn't complain when giving `compose` a callable like that. I think PHP / PHPStan knows the difference between a random array and a callable.

**Review by harmenjanssen — APPROVED** [review]
Great, thanks! LGTM!

**Pull request merged**